tinygrad/helpers.py

```diff
@@ -143,25 +163,29 @@ def is_float(x: DType) -> bool: return x in (dtypes.float16, dtypes.float32, dty
   @staticmethod
   def is_unsigned(x: DType) -> bool: return x in (dtypes.uint8, dtypes.uint16, dtypes.uint32, dtypes.uint64)
   @staticmethod
-  def from_np(x) -> DType: return DTYPES_DICT[np.dtype(x).name]
+  def from_np(x) -> DType:
+    if isinstance(x, DType): return DTYPES_DICT[x.np if x.np is not None else "float64"]
+    if isinstance(x, str): return DTYPES_DICT[x]
+    y: np.dtype = x
+    return DTYPES_DICT[y.name]
   @staticmethod
   def fields() -> Dict[str, DType]: return DTYPES_DICT
-  bool: Final[DType] = DType(0, 1, "bool", np.bool_)
-  float16: Final[DType] = DType(9, 2, "half", np.float16)
+  bool: Final[DType] = DType(0, 1, "bool", "bool", ctypes.c_byte)
+  float16: Final[DType] = DType(9, 2, "half", "float16")
   half = float16
-  float32: Final[DType] = DType(10, 4, "float", np.float32)
+  float32: Final[DType] = DType(10, 4, "float", "float32", ctypes.c_float)
   float = float32
-  float64: Final[DType] = DType(11, 8, "double", np.float64)
+  float64: Final[DType] = DType(11, 8, "double", "float64", ctypes.c_double)
   double = float64
-  int8: Final[DType] = DType(1, 1, "char", np.int8)
-  int16: Final[DType] = DType(3, 2, "short", np.int16)
-  int32: Final[DType] = DType(5, 4, "int", np.int32)
+  int8: Final[DType] = DType(1, 1, "char", "int8", ctypes.c_byte)
+  int16: Final[DType] = DType(3, 2, "short", "int16", ctypes.c_short)
+  int32: Final[DType] = DType(5, 4, "int", "int32", ctypes.c_int)
   int = int32
-  int64: Final[DType] = DType(7, 8, "long", np.int64)
-  uint8: Final[DType] = DType(2, 1, "unsigned char", np.uint8)
-  uint16: Final[DType] = DType(4, 2, "unsigned short", np.uint16)
-  uint32: Final[DType] = DType(6, 4, "unsigned int", np.uint32)
-  uint64: Final[DType] = DType(8, 8, "unsigned long", np.uint64)
+  int64: Final[DType] = DType(7, 8, "long", "int64", ctypes.c_long)
```

Comment:
Why are these strings here?

Reply:
We need the strings so we can support numpy(). They act the same when used in np.array(dtype=) but are not np.dtypes().

Will work on removing them.